public/app/features/alerting/unified/components/rules/RulesTable.tsx

```diff
@@ -146,14 +146,10 @@ function useLazyLoadRulerRules(rules: CombinedRule[]) {
   useEffect(() => {
     if (prometheusRulesPrimary) {
       actions.execute();
-    } else {
-      // We need to reset the actions to update the rules if they changed
-      // Otherwise useAsync acts like a cache and always return the first rules passed to it
-      actions.reset();
     }
-  }, [rules, actions]);
+  }, [actions]);
 
-  return state;
+  return prometheusRulesPrimary ? state : { result: rules, status: 'success' };
```

Comment:
How about wrappiing the second object into a `useMemo`?
Just to keep the reference the same between rerenders